src/exact/ExactMapper.cpp

```diff
@@ -59,6 +59,8 @@ void ExactMapper::map(const MappingSettings& settings) {
 	} while (next_combination(qubits.begin(), qubits.begin()+qc.getNqubits(), qubits.end()));
 
 	// 3) determine exact mapping for this qubit choice
+	if (this->settings.bddLimits>0)	
+		this->settings.bddLimits = findLongestPath(architecture.getCouplingMap(), architecture.getNqubits());
```

Comment:
there should be a flag that enables/disables this feature and a separate variable indicating the limit.